### PR TITLE
KNL-1460 - Allow for null actor (and probably others) in LearningReso…

### PR DESCRIPTION
**Pull request by jonespm**
…urceStore xapi service constructor

### CODE REVIEW

File: kernel/kernel-impl/src/main/java/org/sakaiproject/event/impl/BaseLearningResourceStoreService.java [hunk 1]
```diff
@@ -300,26 +300,43 @@ public void update(Observable observable, Object object) {
...
+    	LRS_Verb verb=null;
+    	LRS_Actor actor=null;
+    	LRS_Context context=null;
+    	LRS_Object object=null;
         if (event.getLrsStatement() != null) {
-            return event.getLrsStatement();
+            statement =  event.getLrsStatement();
+            //If the statement is fully populated nothing left to do
+            if (statement.isPopulated()) {
+				return statement;
```
Comment (relong76) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/event/impl/BaseLearningResourceStoreService.java]: Indentation looks a bit off here...

File: kernel/kernel-impl/src/main/java/org/sakaiproject/event/impl/BaseLearningResourceStoreService.java [hunk 2]
```diff
@@ -300,26 +300,43 @@ public void update(Observable observable, Object object) {
...
-                        statement.setContext(c);
-                    }
-                } else {
-                    statement = null;
-                }
-            } else {
-                statement = null;
+        	//If verb not set try to get it from the event
+        	if (verb ==null) {
+        		verb = getEventVerb(event);
```
Comment (relong76) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/event/impl/BaseLearningResourceStoreService.java]: ditto :)

**Comment by jonespm** [comment]
@relong76 I prettified that method for you.

The main gist of this change was to remove the Exceptions thrown by the constructor and just have it set the value (which doesn't seem to be used much) of populated to be false if it's not fully populated. This seemed like the intention anyway of that value that was already there. Because the service is able to run some methods to try to populate a LRS_Statement based on data in an event. But because of some other changes where you can include an LRS_Statement *in* and event you won't have an event first, the circular problem.

**Comment by relong76** [comment]
Aww, just for ME?? :)

Excellent work, sir!

**Pull request merged**